compiler/resolution/functionResolution.cpp

```diff
@@ -1397,7 +1397,7 @@ bool canCoerceAsSubtype(Type*     actualType,
 
     // Check that the decorators allow coercion
     if (canCoerceDecorators(actualDecorator, formalDecorator,
-                            /*allowNonSubtypes*/ false,
+                            /*allowNonSubtypes*/ true,
```

Comment:
I would expect this PR to instead adjust canCoerceDecorators itself. It looks to me like the `allowNonSubtypes` argument could be removed entirely (and replaced with `true`).